plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1680,6 +1681,14 @@ fc::time_point producer_plugin_impl::calculate_block_deadline( const fc::time_po
    }
 }
 
+bool producer_plugin_impl::start_block_interrupted( const fc::time_point& deadline ) {
+   if( _pending_block_mode == pending_block_mode::speculating ) {
+      return _received_block;
+   } else {
+      return deadline <= fc::time_point::now();
+   }
+}
+
```

Comment:
I don't think this is correct since the deadline if considered only when `_pending_block_mode` is *not* speculating.

Reply:
maybe:
```
return (_pending_block_mode == pending_block_mode::speculating && _received_block) ||
        deadline <= fc::time_point::now();
```

Reply:
That is what we want. A speculative block doesn't have a deadline. It was given one according to scheduled block, but that is really not meaningful for a speculative block.

Reply:
We could argue that this is too much of a change for 3.1 & 3.2 and still go with the meaningless deadline and only remove the deadline for `main`.

Reply:
OK, thanks Kevin. Just a detail, which probably doesn't matter here, but personally I try to avoid some paths with no return, even if they are unreachable, so I would do either:

```
   if( _pending_block_mode == pending_block_mode::speculating ) {
      return _received_block;
   } 
   return deadline <= fc::time_point::now();
```

or 

```
return _pending_block_mode == pending_block_mode::speculating ? _received_block : deadline <= fc::time_point::now();
```

Reply:
Done.